src/e2e/puppeteer/helpers/waitForFrames.ts

```diff
@@ -1,23 +1,28 @@
 import { page } from '../setup'
 
 /**
- * Waits for two consecutive animation frames to ensure complete rendering.
+ * Waits for a given number of pairs of consecutive animation frames to ensure complete rendering.
  *
  * This technique guarantees:
  * 1. All pending `requestAnimationFrame` callbacks execute (first frame).
  * 2. Resulting DOM updates process through style/layout/paint (second frame).
  * 3. Browser reaches stable visual state for screenshots.
  *
  * Essential for eliminating flaky tests where:
- * - Components use rAF for state updates (e.g., Superscript calculations).
+ * - Components use rAF for state updates.
  * - CI environments have slower frame rates (20-30 FPS vs local 60 FPS).
  * - Fixed timeouts (e.g., `sleep(200)`) fail to capture final render.
  */
-const waitForFrames = () =>
-  page.evaluate(() => {
-    return new Promise(resolve => {
-      requestAnimationFrame(() => requestAnimationFrame(resolve))
-    })
-  })
+const waitForFrames = (count: number = 1): Promise<void> => {
```

Comment:
The name is a bit misleading, since each invocation waits two frames. e.g. `waitForFrames(2)` actually waits 4 frames.

I think it would be better to keep it 1:1 for comprehensibility, and not do it in pairs.